pennylane/circuit_drawer/drawable_layers.py

```diff
@@ -0,0 +1,130 @@
+        occupied_wires_per_layer (list[set[int]]): which wires will already have something drawn
+            on them
+
+    Returns:
+        int: layer to place relevant operation in
+    """
+
+    if occupied_wires_per_layer[checking_layer] & op_occupied_wires:
+        # this layer is occupied, use higher one
+        return checking_layer + 1
+    if checking_layer == 0:
+        # reached first layer, so stop
+        return 0
+    # keep pushing the operation back
+    return _recursive_find_layer(checking_layer - 1, op_occupied_wires, occupied_wires_per_layer)
+
+
+def drawable_layers(ops, wire_map=None):
+    """Determine non-overlapping yet dense placement of operations for drawing.
+
+    Args:
+        ops Iterable[~.Operator]: a list of operations
+
+    Keyword Args:
+        wire_map=None (dict): a map from wire label to non-negative integers
+
+    Returns:
+        list[set[~.Operator]] : Each index is a set of operations
+            for the corresponding layer
+    """
+
+    if wire_map is None:
+        wire_map = default_wire_map(ops)
+
+    # initialize
+    max_layer = 0
+    occupied_wires_per_layer = [set()]
+    ops_per_layer = [set()]
+
+    # loop over operations
+    for op in ops:
+        if len(op.wires) == 0:
+            # if no wires, then it acts of all wires
+            # for example, state and sample
+            mapped_wires = set(wire_map.values())
+            op_occupied_wires = mapped_wires
+        else:
+            mapped_wires = {wire_map[wire] for wire in op.wires}
+            op_occupied_wires = set(range(min(mapped_wires), max(mapped_wires) + 1))
+
+        op_layer = _recursive_find_layer(max_layer, op_occupied_wires, occupied_wires_per_layer)
+
+        # see if need to add new layer
+        if op_layer > max_layer:
+            max_layer += 1
+            occupied_wires_per_layer.append(set())
+            ops_per_layer.append(set())
+
+        # Add to op_layer
+        ops_per_layer[op_layer].add(op)
+        occupied_wires_per_layer[op_layer].update(op_occupied_wires)
+
+    return ops_per_layer
+
+
+def drawable_grid(ops, wire_map=None):
+    """Determine non-overlapping yet dense placement of operations for drawing.  Returns
+    structure compatible with ``qml.circuit_drawer.Grid``.
+
+    Args:
+        ops Iterable[~.Operator]: a list of operations
+
+    Keyword Args:
+        wire_map=None (dict): a map from wire label to non-negative integers
+
+    Returns:
+        List[List[~.Operator]] : layers compatible with grid objects
+    """
+    if wire_map is None:
+        wire_map = default_wire_map(ops)
+
+    if len(ops) == 0:
+        if len(wire_map) == 0:
+            return [[]]
+        return [[] for _ in range(len(wire_map))]
+
+    ops_per_layer = drawable_layers(ops, wire_map=wire_map)
+
+    n_wires = len(wire_map)
+    n_layers = len(ops_per_layer)
+
+    grid = [[None for _ in range(n_layers)] for _ in range(n_wires)]
```

Comment:
We are creating the grid here as some spots have not been occupied when calling `drawable_layers`, right? Maybe worth leaving a comment.

Reply:
This is just a conversion between the data type created by `drawable_layers` and that needed for the existing circuit drawer.

I hope I will be able to later refractor this away, but that will take multiple steps.